actions/post_actions.jsx

```diff
@@ -99,57 +99,64 @@ function dispatchPostActions(post, websocketMessageProps) {
     dispatch(sendDesktopNotification(post, websocketMessageProps));
 }
 
-export async function flagPost(postId) {
-    await PostActions.flagPost(postId)(dispatch, getState);
-
-    const rhsState = getRhsState(getState());
-
-    // This is a hack that should be fixed with better reducers/actions, see MM-9793
-    if (rhsState === RHSStates.FLAG) {
-        let results = getState().entities.search.results;
-        const index = results.indexOf(postId);
-        if (index === -1) {
-            results = [...results, postId];
-
-            const posts = {};
-            results.forEach((id) => {
-                posts[id] = Selectors.getPost(getState(), id);
-            });
-
-            results.sort((a, b) => comparePosts(posts[a], posts[b]));
-
-            dispatch({
-                type: SearchTypes.RECEIVED_SEARCH_POSTS,
-                data: {posts, order: results},
-            });
+export function flagPost(postId) {
+    return async (doDispatch, doGetState) => {
```

Comment:
It might be nice to remove to get rid of the global `dispatch` and `getState` fields in this file so that we don't need to worry about shadowing them or calling them accidentally

Reply:
Yup, I thought to do it last after migrating all inside the post actions.